service/dap/server.go

```diff
@@ -0,0 +1,420 @@
+		//s.onThreadsRequest(request)
+	case *dap.TerminateThreadsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onTerminateThreadsRequest(request)
+	case *dap.EvaluateRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onEvaluateRequest(request)
+	case *dap.StepInTargetsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onStepInTargetsRequest(request)
+	case *dap.GotoTargetsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onGotoTargetsRequest(request)
+	case *dap.CompletionsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onCompletionsRequest(request)
+	case *dap.ExceptionInfoRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onExceptionInfoRequest(request)
+	case *dap.LoadedSourcesRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onLoadedSourcesRequest(request)
+	case *dap.DataBreakpointInfoRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onDataBreakpointInfoRequest(request)
+	case *dap.SetDataBreakpointsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onSetDataBreakpointsRequest(request)
+	case *dap.ReadMemoryRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onReadMemoryRequest(request)
+	case *dap.DisassembleRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onDisassembleRequest(request)
+	case *dap.CancelRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onCancelRequest(request)
+	case *dap.BreakpointLocationsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+		//s.onBreakpointLocationsRequest(request)
+	default:
+		// This should have been returned as an error from
+		// dap.ReadProtocolMessage(), so we should never reach here.
+		panic(fmt.Sprintf("Unable to process %#v", request))
+	}
+}
+
+func (s *Server) send(message dap.Message) {
+	jsonmsg, _ := json.Marshal(message)
+	s.log.Debug("[-> to client]", string(jsonmsg))
+	dap.WriteProtocolMessage(s.conn, message)
+}
+
+func (s *Server) onInitializeRequest(request *dap.InitializeRequest) {
+	// TODO(polina): Respond with an error if debug session is in progress?
+	response := &dap.InitializeResponse{Response: *newResponse(request.Request)}
+	response.Body.SupportsConfigurationDoneRequest = true
+	// TODO(polina): support this to match vscode-go functionality
+	response.Body.SupportsSetVariable = false
+	s.send(response)
+}
+
+func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
+	// TODO(polina): Respond with an error if debug session is in progress?
+	program, ok := request.Arguments["program"]
+	if !ok || program == "" {
+		s.sendErrorResponse(request.Request,
+			3000, "Failed to launch",
+			"The program attribute is missing in debug configuration.")
+		return
+	}
+	s.config.ProcessArgs = []string{program.(string)}
+	s.config.WorkingDir = filepath.Dir(program.(string))
+	// TODO: support program args
+
+	stop, ok := request.Arguments["stopOnEntry"]
```

Comment:
```
if stop, ok := ...; ok {
  s.stopOnEntry = stop
}
```

Reply:
It's more idiomatic in Go to do the `if stuff...; check stuff` clause, especially because you're not using either `stop` or this `ok` outside the assignment to `s.stopOnEntry`.

This is a nit. If you feel your version is more readable, keep it as is, no worries.